pkg/webui/console/views/device-general-settings/network-server-form/index.js

```diff
@@ -348,6 +374,16 @@ const NetworkServerForm = React.memo(props => {
       <MacSettingsSection activationMode={initialActivationMode} isClassB={isClassB} />
       <SubmitBar>
         <Form.Submit component={SubmitButton} message={sharedMessages.saveChanges} />
+        <ModalButton
+          type="button"
+          warning
+          naked
+          message={m.resetTitle}
+          modalData={{
+            message: m.modalMessage,
+          }}
+          onApprove={handleMacReset}
+        />
```

Comment:
Let's add this as a regular button within the form rather. I'm afraid that people will miss or misinterpret the button when it is part of the submit bar. Also, using a regular field will enable us to use a tooltip description.

You could use a `<Form.InfoField />` for this. It should have the label `Session and MAC state reset`.

Additionally, please add a tooltip that reads as follows:

> Resetting the session context and MAC state will reset the end device to its initial (factory) state. This includes \<b>resetting the frame counters\<b/> and any other persisted MAC setting on the end device. Over The Air Activation (OTAA) end devices will also lose their session context. This means that such devices \<b>must rejoin\</b> the network to continue operation. Activation-by-personalization (ABP) end devices will only reset the MAC state, while preserving up/downlink queues.

Again, @htdvisser, @adriansmares please double check the wording here.

Additionally, in this case, we should use a field description reading:

> You can use this e.g. to cause a frame counter reset